File: packages/shared/src/types/review.ts
import { Timestamp } from 'firebase/firestore';

/**
 * ReviewProps — a customer product review document (collection `reviews`), written by the user app
 * after an order is Delivered (spec §1.11). The admin panel only reads, toggles publish, adds/removes
 * an admin response, or hard-deletes — it never creates reviews.
 *
 * WHY shared (not admin-local): the user app writes these docs and Cloud Functions recompute product
 * `averageRating`/`totalReviews` from them, so the shape must be identical everywhere.
 * WHY `keywords`: a lowercased token array the user app populates so admin search can use a single
 * `array-contains` query (no full-text service). Date fields are Firestore `Timestamp` (never epoch ms).
 */
export interface ReviewProps {
  id: string;
  userId: string;
  userName: string;
  userImage: string;
  productId: string;
  productName: string;
  productImage: string;
  variantId: string;
  variantName: string;
  orderId: string;
  rating: number;
  title: string;
  comment: string;
  photos: string[];
  isPublished: boolean;
  isVerifiedPurchase: boolean;
  adminResponse: string;
  keywords: string[];
  createdAt: Timestamp;
  updatedAt: Timestamp;
}

File: apps/app/lib/features/reviews/domain/entities/review.dart
import 'package:equatable/equatable.dart';

/// A product review document (`reviews/{id}`). Written by the app after an order
/// is Delivered; the admin only toggles publish / responds / deletes.
class Review extends Equatable {
  final String id;
  final String userId;
  final String userName;
  final String userImage;
  final String productId;
  final String productName;
  final String productImage;
  final String variantId;
  final String variantName;
  final String orderId;
  final double rating;
  final String title;
  final String comment;
  final List<String> photos;
  final bool isPublished;
  final bool isVerifiedPurchase;
  final String adminResponse;
  final List<String> keywords;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const Review({
    required this.id,
    required this.userId,
    required this.userName,
    required this.userImage,
    required this.productId,
    required this.productName,
    required this.productImage,
    required this.variantId,
    required this.variantName,
    required this.orderId,
    required this.rating,
    required this.title,
    required this.comment,
    required this.photos,
    required this.isPublished,
    required this.isVerifiedPurchase,
    required this.adminResponse,
    required this.keywords,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        userName,
        userImage,
        productId,
        productName,
        productImage,
        variantId,
        variantName,
        orderId,
        rating,
        title,
        comment,
        photos,
        isPublished,
        isVerifiedPurchase,
        adminResponse,
        keywords,
        createdAt,
        updatedAt,
      ];
}
